api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -56,19 +60,28 @@ public void bufferBillingProject() {
 
   public void syncBillingProjectStatus() {
     BillingProjectBufferEntry entry = billingProjectBufferEntryDao
-        .findFirstByStatusOrderByCreationTimeAsc(StorageEnums.billingProjectBufferStatusToStorage(CREATING));
+        .findFirstByStatusOrderByLastSyncRequestTimeAsc(StorageEnums.billingProjectBufferStatusToStorage(CREATING));
 
     if (entry == null) {
       return;
     }
 
-    switch (fireCloudService.getBillingProjectStatus(entry.getFireCloudProjectName()).getCreationStatus()) {
-      case READY:
-        entry.setStatusEnum(AVAILABLE);
-        break;
-      case ERROR:
-        entry.setStatusEnum(ERROR);
-        break;
+    entry.setLastSyncRequestTime(new Timestamp(clock.instant().toEpochMilli()));
+
+    try {
+      switch (fireCloudService.getBillingProjectStatus(entry.getFireCloudProjectName())
+          .getCreationStatus()) {
+        case READY:
+          entry.setStatusEnum(AVAILABLE);
+          break;
+        case ERROR:
+          log.warning(String.format("SyncBillingProjectStatus: BillingProject %s creation failed",
+              entry.getFireCloudProjectName()));
+          entry.setStatusEnum(ERROR);
+          break;
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
nit: Would probably be more consistent to write this exception via the logger, e.g. `log.log(Level.WARNING, "Firecloud project check failed", e);`